Bloom.md

```diff
@@ -0,0 +1,341 @@
+This approach was not chosen because we can already save and restore bloom object data from RDB data using the RESTORE command
+without having to support additional Module commands.
+
+### Migrating workloads from ReBloom:
+
+Customers that currently use ReBloom can move to ValkeyBloom using two approaches:
+
+1. Create the bloom filter objects to have the same properties using BF.RESERVE / BF.INSERT on Valkey (with ValkeyBloom loaded).
+Re-populate the bloom filter objects by inserting items by moving the existing bloom workload to the Valkey server
+(with ValkeyBloom loaded). The workload can be moved without any errors since we are API Compatible.
+
+Pros
+* This is the simplest option in terms of effort - assuming the user is fine with recreating bloom objects and adding
+    items on the Valkey server (with ValkeyBloom).
+
+Cons
+* The user will need to re-create the bloom objects (using BF.RESERVE/BF.INSERT) and populate these objects **AFTER**
+    switching to Valkey (with ValkeyBloom).
+
+2. Users can generate an AOF file from a server (that has the ReBloom module loaded) and with an on-going bloom workload
+that creates bloom filters & inserts items into them. Next, this can be re-played on a Valkey Server (with ValkeyBloom loaded).
+Then, the user can move their existing bloom workload to the Valkey server (with ValkeyBloom loaded).
+
+Pros
+* Bloom objects can be created on the user's existing system (with Rebloom) and the user can also populate the objects
+    by adding items (BF.ADD/BF.MADD) **BEFORE** switching to Valkey (with ValkeyBloom).
+
+Cons
+* The user will need to manage AOF file generation on the server (with ReBloom) and ensure that the contents do not get
+    re-written (e.g. as a result of BGREWRITEAOF) into a RDB file. This is because the RDB file with bloom object data
+    generated by ReBloom is not compatible with ValkeyBloom. The main drawback here is that the AOF can exceed a size
+    limit and get truncated / re-written.
+
+### Memory Management
+
+The bloom data type also supports memory management related callbacks:
+
+* free: Frees the bloom filter object when it is deleted, expired or evicted.
+* defrag: Supports active defrag for Bloom filter objects
+* mem_usage: Reports bytes used by a Bloom object and is used by the MEMORY USAGE command
+* copy: Supports a deep copy of bloom filter objects and is used by the COPY command
+* free_effort: Determine whether the bloom object's memory needs to be lazy reclaimed or synchronously freed. We return
+    the number of filters in the bloom object as the free effort and this is similar to how the core handles free_effort
+    for aggregated objects.
+
+
+### Replication
+
+Every Bloom Filter based write operation (bloom object creations, scaling, setting of an item on a filter which returns 1
+indicating a new entry, etc) will be replicated to replica nodes. Attempts of adding an already existing item to a bloom
+object (which returns 0) will not replicated.
```

Comment:
The configuration mentioned below can vary across primary/replica. How will it affect the replication stream then?

Reply:
For e.g., if the config value for `bf.bloom_fp_rate`  is set differently on primary and replica. Operation like `BF.ADD` might succeed on primary and fail on replica and vice versa.

Reply:
On this point, I think we should remove external facing configuration and set default value in the code and let it be overridable for each bloom filter key. WDYT ?

Reply:
We would expect the same default `bf.bloom_fp_rate` value to be set across the cluster. If this is not the case, it is more likely that adding of an item on the primary succeeds and does not succeed on the replica (false positive indicating that the item already exists).   

Through commands, it is possible to specify the false positive rate through the BF.INSERT and BF.RESERVE commands. BF.ADD and BF.MADD only accept items as arguments. If a user directly starts using BF.ADD, then it will be created using the default false positive rate.

This is fine assuming the user is OK with the default properties. A config will help to override this on a globally for the node. But I understand the concern and also understand the alternative (using BF.INSERT/BF.RESERVE)